airflow/executors/kubernetes_executor_types.py

```diff
@@ -0,0 +1,35 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple
+
+if TYPE_CHECKING:
+    from airflow.executors.base_executor import CommandType
+    from airflow.models.taskinstance import TaskInstanceKey
+
+    # TaskInstance key, command, configuration, pod_template_file
+    KubernetesJobType = Tuple[TaskInstanceKey, CommandType, Any, Optional[str]]
+
+    # key, pod state, pod_name, namespace, resource_version
+    KubernetesResultsType = Tuple[TaskInstanceKey, Optional[str], str, str, str]
+
+    # pod_name, namespace, pod state, annotations, resource_version
+    KubernetesWatchType = Tuple[str, str, Optional[str], Dict[str, str], str]
+
+ALL_NAMESPACES = "ALL_NAMESPACES"
+POD_EXECUTOR_DONE_KEY = "airflow_executor_done"
```

Comment:
Only issue I have with this is these two doesn’t seem to fit in this module.

Reply:
Thanks for the review!

I needed a "safe to import" module to store the types and constants for both `kubernetes_executor.py` and  `kubernetes_executor_utils.py` to import from. It allows it to be a global import. I called this module types since it's mostly that, and the constants felt pretty close in spirit.

Some paths forward:

1. Leave it as it is now and don't complicate it further for just two constants
1. Create a separate module for constants and put these two in there
1. Move the constants to `kubernetes_executor_utils.py` and then locally import them in the places they're required in `kubernetes_executor.py`

I'm inclined to just stick with 1. Let me know what you prefer @uranusjr, or if you have any other options you think are preferred.

Reply:
Maybe just rename this module to something that says e.g. `constants` instead of `types`? The problem is not that these values can’t be with other values here, but that these two are not types.

Reply:
> Maybe just rename this module to something that says e.g. constants instead of types?

But then you just have the same problem but on the other end, i.e. the Types in this module will no longer "fit in this module", right? I don't understand why that is better than it is now.

Reply:
Type aliases kind of feel like constants to me but maybe I’m just weird. My top preference is definitely to split those into two files.

Reply:
Sure, I can do that split in the follow PR